Ladder step4 1

junibibi opened the pull request:
4-1 사다리게임 프로그램 설계하기를 진행하였습니다.

JurdyPark committed:
이중배열생성

사다리가 연결되는지 안되는지 저장하는 bool이중배열 생성

JurdyPark committed:
이중배열에 bool	값 넣음

랜덤한 값을 bool값으로 바꿔서 이중배열에 넣어줬음

JurdyPark committed:
이중배열 원소범위, for문 범위 수정

이중배열 원소갯수 부분 수정
for문 범위를 수정

JurdyPark committed:
사다리 있고없음을 표시하는 배열 생성

이중배열에 들어있는 Bool값을 String타입으로 바꿔서 이중배열에 넣음

JurdyPark committed:
사다리를 출력하는 함수와 사다리게임함수 호출

그림으로 배열에 있는 사다리를 출력하는 함수.
사다리게임 호출

불필요하게 생성한 변수 삭제
범위를 숫자->Int변수명으로 수정

JurdyPark committed:
사다리게임 호출부분 수정

사다리게임 호출을 for문 안에 넣어줌

JurdyPark committed:
인원수와 사다리갯수를 사용자가 입력하게끔 수정함

인원수와 사다리갯수를 사용자가 입력하게끔 수정함
print 주석처리함

JurdyPark committed:
변수명 수정

애매한 변수명 수정함

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -8,3 +8,66 @@
+
+//ladder그림을 받을 이중배열 생성
+var ladderArr : [[String]] = Array(repeating: Array(repeating: "-", count: peopleNum-1), count: ladderNum)
+
```
여기 작성한 코드 흐름을 역할 단위별로 함수를 구분해보세요.
아마 줄바꿈을 한 곳들이 문단을 나눈 것처럼 함수 단위로 구분할 수 있을 것 같습니다.
입력과 출력이 있는 함수로 기능을 구현하는 것을 연습해야 합니다. 
그리고 맨 아래쪽에는 구현한 함수들을 입력 - 처리 - 출력 순서대로 조합해서 호출하는 최상위 함수를 만들어보세요.

JurdyPark committed:
사용자한테 입력받는 부분 함수로 구현

입력부분을 함수로 구현했습니다

JurdyPark committed:
랜덤한 수를 Bool값으로 변환해서 저장하는 함수

핸덤한 숫자를 받아 Bool값으로 변환하는 함수를 구현하였고
함수 밖에서 생성했던 이중배열을 함수안에서 생성하도록 수정했습니다.

JurdyPark committed:
Bool값을 사다리모양 String으로 변환하여 저장하는 함수

Bool로 저장된 값을 "-"와 " "으로 변환하여 저장하는 함수 구현하고
함수밖에 생성했던 배열을 함수 안에서 생성하도록 수정하였습니다.

JurdyPark committed:
배열에 있는 그림으로 사다리1행을 만드는 함수 구현

시작할때 세로줄을 하나 추가해 주고, 배열에 있는 그림을 하나씩 가져올때마다 다음에 세로줄을 추가해서 사다리 1행을 만드는 함수를 만들었습니다.

JurdyPark committed:
사다리높이만큼 사다리그림을 1행씩 증가시키는 함수구현.

원하는 높이만큼 사다리행을 증가시키는 함수 만들었습니다

JurdyPark committed:
게임실행하는 함수 구현

입력 - 처리 - 출력 순서대로 조합해서 호출하는 최상위 함수를 만들었습니다.

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -8,66 +8,83 @@
 import Foundation
 
+var peopleNum = 0
+var ladderNum = 0
```
아직 전역 변수가 남아있네요. 함수가 외부에 있는 변수에 의존하는 것은 좋지 않습니다.
입력을 처리하는 함수가 return 하도록 만들어보세요.

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -8,66 +8,83 @@
 
+//랜덤한숫자를 받아서 Bool값으로 저장하는 함수
+func randomNumChangeBoolArray(ladderHeight: Int, ladderWidth:Int)->[[Bool]]{
+    var boolArr : [[Bool]] = Array(repeating: Array(repeating: true,count:peopleNum-1 ), count: ladderNum)
```
변수 이름이나 함수 이름에 타입 정보를 붙이지 말고,
대신에 변수나 함수의 역할, 의도를 표현하는 단어만 사용해보세요. 타입은 역할에 따라 결정될 수 있습니다.

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -8,66 +8,83 @@
-            ladderArr[ladderArrIndex][ladderArrInArrIndex]=(" ")
+//이중배열에 들어있는 Bool값을 그림으로 전환하여 저장하는 함수
+func boolArrayChangeStringArray(ladderHeight: Int, ladderWidth:Int, boolArray: [[Bool]])->[[String]]{
+    var ladderArr : [[String]] = Array(repeating: Array(repeating: "-", count: peopleNum-1), count: ladderNum)
```
bool, Array, Number, String 처럼 타입을 의미하는 단어를 제외해보세요.

godrm reviewed (commented):
지금 정도 수준의 함수 분리는 좋습니다. 
코드로 정확한 의도를 드러내기 위한 표현을 써보도록 하세요.

JurdyPark committed:
유저가 입력하는 사람수와 사다리수를 받는 함수 수정

전역변수를 없애고 입력한 사람수와 사다리수를 리턴하도록 수정했습니다.

JurdyPark committed:
랜덤한숫자를 받아서 Bool값으로 저장하는 함수 수정.

입력함수를 파라미터로 받아서 리턴값을 이용해 함수가 실행되도록 수정하였습니다.

JurdyPark committed:
이중배열에 들어있는 Bool값을 그림으로 저장하는 함수 수정.

전역변수로 값을 받았던 부분을 파라미터로 받는 Bool배열을 이용하여 받도록 수정하였습니다.

JurdyPark committed:
배열에있는 그림으로 사다리1행 만드는 함수 수정.

전역변수로 받았던 값을 내부에서 받도록 수정했습니다.

JurdyPark committed:
사다리높이만큼 사다리그림을 1행씩 증가시키는 함수, 게임실행 함수 수정.

전역변수로 받는 값을 내부에서 받도록 수정하였습니다.
게임실행하는 함수의 파라미터를 수정하였습니다.

JurdyPark committed:
변수와 함수이름 수정.

변수와 함수이름에 타입정보를 붙이지 않고 역할과 의도를 표현하여 수정했습니다.
